js/twister_formatpost.js

```diff
@@ -379,196 +404,228 @@ function htmlFormatMsg(msg, mentions) {
+    var stopCharsRightHashtags = '>/\\.,:;?!%\'"[](){}^|«»…\u201C\u201D\u2026\u2014\u4E00\u3002\uFF0C\uFF1A\uFF1F\uFF01\u3010\u3011\u2047\u2048\u2049'  // same as stopCharsTrailing but without '*~_-`' plus '>'
         + whiteSpaces;
-    var stopCharsMarkDown = '/\\*~_-`.,:;?!%+=&\'"[](){}^|«»…\u201C\u201D\u2026\u2014\u4E00\u3002\uFF0C\uFF1A\uFF1F\uFF01\u3010\u3011';
-    var i, j, k, str, strEncoded;
-    var html = [];
-
-    msg = markdown(escapeHtmlEntities(msg),
-        '`', 'samp');  // kind of monospace, sequence of chars inside will be escaped from markup
-    for (i = 0; i < msg.length - 7; i++) {
-        var htmlPiece = undefined;
-        var msgSliceIdx;
-        if (msg.slice(i, i + 2) === '](') {
-            // FIXME there can be text with [] inside [] or links with () wee need to handle it too
-            j = getStrStart(msg, i - 1, '[', true, '');
-            if (j < i) {
-                k = getStrEnd(msg, i + 2, ')', true, '');
-                if (k > i + 1) {
-                    var a = $('#external-page-link-template')[0].cloneNode();
-                    a.href = proxyURL(msg.slice(i + 2, k + 1));
-                    a.text = msg.slice(j, i);
-                    htmlPiece = a.outerHTML;
-                    msgSliceIdx = [j - 1, k + 2];
+    var stopCharsMarkout = '/\\*~_-`.,:;?!%+=&\'"[](){}^|«»…\u201C\u201D\u2026\u2014\u4E00\u3002\uFF0C\uFF1A\uFF1F\uFF01\u3010\u3011\u2047\u2048\u2049';
+    var i, j, k, str;
+
+    msg = {str: escapeHtmlEntities(msg), htmlEntities: []};
+
+    msg = markout(msg, '`', 'samp');  // <samp> tag is kind of monospace, here sequence of chars inside it will be escaped from markup
+
+    // handling links
+    for (i = 0; i < msg.str.length - 7; i++) {
+        if (msg.str.slice(i, i + 2) === '](') {
+            // FIXME there can be text with [] inside [] or links with () we need to handle it too
+            j = getSubStrStart(msg.str, i - 2, '[', true, '');
+            if (j < i - 1) {
+                k = getSubStrEnd(msg.str, i + 3, ')', true, whiteSpaces);
+                if (k > i + 2) {
+                    var linkName = msg.str.slice(j, i);  // name of possiible link
+                    for (i += 2; i < k; i++) {
+                        if (whiteSpacesUrl.indexOf(msg.str[i]) === -1)  // drop whitespaces and ' and "  // apostrophes and quotes to prevent injection of js events
+                            break;
+                    }
+                    if (i < k) {
+                        var x = getSubStrEnd(msg.str, i, ':', false, '') + 1;
+                        // following check is NOT for real protection (we have blocking CSP rule instead), it's just to aware people
+                        if (msg.str[i] === '#' || (x > i && x < k && (msg.str.slice(x - 6, x).toLowerCase() === 'script'  // other things would be added when W3C and all the people invent it
```

Comment:
agreed, it would not be useful for XSS in twister.
anyway just crossed with that interesting page and i thought it would be worth mentioning this kind of trick (lots of them actually).